core/vm/lfvm/converter.go

```diff
@@ -13,26 +13,52 @@ type cache_key struct {
 	contract_length int
 }
 
+type cache_val struct {
+	oldCode []byte
+	code    Code
+}
+
+var changedAddress01 = common.HexToAddress("0xA7CC236F81b04c1058e9bfb70E0Ee9940e271676")
+var changedAddress02 = common.HexToAddress("0xAD0FB83a110c3694faDa81e8B396716a610c4030")
+var changedAddress03 = common.HexToAddress("0xA8B3C9f298877dD93F30E8Ed359956faE10E8797")
+
 var mu = sync.Mutex{}
-var cache = map[cache_key]Code{}
+var cache = map[cache_key]cache_val{}
 
-func Convert(addr common.Address, code []byte, with_super_instructions bool) (Code, error) {
+func Convert(addr common.Address, code []byte, with_super_instructions bool, blk uint64, create bool) (Code, error) {
 	key := cache_key{addr, len(code)}
 	mu.Lock()
 	res, exists := cache[key]
-	if exists {
-		mu.Unlock()
-		return res, nil
+	if exists && !create {
+		isEqual := true
+		if addr == changedAddress01 || addr == changedAddress02 || addr == changedAddress03 {
+			// fmt.Println("Address: ", addr.String(), " blk: ", blk)
+
+			for i, v := range res.oldCode {
+				if v != code[i] {
```

Comment:
nit: the access `code[i]` may cause an out-of-bound access error if the new code is shorter than the old code (might not happen); to be safe, and more explicit and what is done, maybe use `bytes.Equal(res.oldCode, code)`?

feel free to ignore

Reply:
I understand your point, but it goes in this if branch only when code with same lenght exist in the cache, so compared arrays have same length every time

Reply:
Right, missed that the length is in the cache key.